spec/command_line_ui_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'command_line_ui'
+require 'board'
+require 'player_symbols'
+require 'player_symbols'
+
+RSpec.describe CommandLineUI do
+
+  before (:each) do
+    @reader_spy = instance_double("PromptReader")
+    @writer_spy = instance_double("PromptWriter")
+  end
+
+  it "reads move from user" do
+    expect(@writer_spy).to receive(:ask_for_next_move)
+    expect(@writer_spy).to receive(:show_board)
+    expect(@reader_spy).to receive(:get_input).and_return("1")
+
+    command_line_ui = CommandLineUI.new(@writer_spy, @reader_spy)
+
+    expect(command_line_ui.get_move_from_player(Board.new)).to be 1 
+  end
+
+  it "reprompts move from user when an alpha character entered" do
+    expect(@writer_spy).to receive(:ask_for_next_move).twice
```

Comment:
These tests look brittle with a lot of dynamic mock setup and a lot of assertions on messages being passed. How do you feel about these tests?